exonum-light-client/src/test/java/com/exonum/client/ExonumHttpClientTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.exonum.client;
+
+import static com.exonum.binding.common.crypto.CryptoFunctions.ed25519;
+import static com.exonum.binding.common.serialization.json.JsonSerializer.json;
+import static com.exonum.client.ExonumHttpClient.HEX_ENCODER;
+import static com.exonum.client.ExonumUrls.SUBMIT_TRANSACTION;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.exonum.binding.common.crypto.KeyPair;
+import com.exonum.binding.common.hash.HashCode;
+import com.exonum.binding.common.message.TransactionMessage;
+import java.io.IOException;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ExonumHttpClientTest {
+  private MockWebServer server;
+  private ExonumClient exonumClient;
+
+  @BeforeEach
+  void start() throws IOException {
+    server = new MockWebServer();
+    server.start();
+
+    exonumClient = ExonumClient.newBuilder()
+        .setExonumHost(server.url("/").url())
+        .build();
+  }
+
+  @AfterEach
+  void shutdown() throws IOException {
+    server.shutdown();
+  }
+
+  @Test
+  void submitTransactionTest() throws InterruptedException {
```

Comment:
If we keep both, I wonder how we can name them to clearer express the difference. submitTransactionResponseTest and submitTransactionRequestTest?

Reply:
Created one integration test